pythonlib/src/mill/pythonlib/PythonModule.scala

```diff
@@ -49,7 +61,7 @@ trait PythonModule extends Module with TaskModule { outer =>
    */
   def transitiveSources: T[Seq[PathRef]] = Task {
     val upstreamSources = Task.traverse(moduleDeps)(_.transitiveSources)().flatten
-    sources() ++ upstreamSources
+    sources() ++ resources() ++ upstreamSources
```

Comment:
It feels a bit strange if `resources()` is included in `transitiveSources()` but then isn't included in `sources()`. 

Come to think of it, do we actually need a `resources()` task? Could we include a `resouces/` directory in the default `sources()` task? I'm not sure which would be the "best" (aka least surprising) behavior here. @lihaoyi any thoughts on this?

Reply:
i think just like we added `transitiveSources` in the we need to add the resources to the JVM resource pool this will also allow us to use the `importlib.resource` 

also @jodersky sir tell me one thing why i we get error of `foo is not a package` if we do something like this 
`importlib.resource.open_file(package='foo', resource='line-count.txt')`

generally when we get `___ is not a package` this means that this folder don't include `__init__.py` or its not included in the `PYTHONPATH` and in our case we have used the `PYTHONPATH` Method but still we are getting same error 

any Thoughts on this ???

Reply:
I don't actually know which way would be better, but let's keep `sources` and `resources` separate tasks for now. I imagine there will be things people want to do to sources that they do not want to do to resources, e.g. you want to typecheck your sources but you do not want to typecheck your resources, or if you want to obfuscate your python sources you probably do not want to obfuscate random `.py` files in resources. 

While this stuff can generally be worked around on a case-by-case basis, I think having `sources` and `resources` be separate is worth maintaining

Reply:
@himanshumahajan138 you need to figure out what `importlib.resources` and `pkgutil.get_data` work, do what needs to be done, and explain to use why it needs to be done that way

Reply:
As part of #3998, I've reworked the way the pythonpath is created.

https://github.com/com-lihaoyi/mill/blob/abf23050ff9e659cbf4fee0d80486b19e504bb6a/pythonlib/src/mill/pythonlib/PythonModule.scala#L51-L106

This hierarchy could be used to include resources as well (see the TODO comment on line 81 of the snippet)

Reply:
yaa i checked and was also doing the same thing by coping your code from commits 😅

Reply:
FIXED!